SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverRPC.cpp

```diff
@@ -196,9 +198,13 @@ struct FSpatialNetDriverRPC::RAIIUpdateContext : FStackOnly
 			FSpatialGDKSpanId SpanId;
 			if (RPCSystem.EventTracer != nullptr)
 			{
-				SpanId = RPCSystem.EventTracer->TraceEvent(
-					FSpatialTraceEventBuilder::CreateMergeSendRPCs(Update.EntityId, Update.Update.component_id),
-					/* Causes */ Update.Spans.GetData()->GetConstId(), /* NumCauses */ Update.Spans.Num());
+				TArray<FSpatialGDKSpanId>& Causes = Update.Spans;
+
+				SpanId = RPCSystem.EventTracer->TraceEvent(MERGE_SEND_RPCS_EVENT_NAME, "", Causes.GetData()->GetConstId(), Causes.Num(),
```

Comment:
It seems quite a common pattern to have an empty string, worth making an overload without the string, saves you a few unneeded string objects.

Reply:
What objects are you referring to?

To create a Trace_Event inside TraceEvent, we will need a valid char *. The function you suggest will require passing a string literal to the creation of the Trace_Event. Would there be a saving here? I can see a saving in terms of not having to pass the pointer into the function. We might also save some read only memory depending on how badly the system stores string literals.

Reply:
What I mean is that this is a lot of work for an empty string, we create an FString temporary and std::string temporary and then create a c_str, out of that.
```
FUserSpanId USpatialEventTracerUserInterface::TraceEvent(UObject* WorldContextObject, const FString& EventType, const FString& EventMessage, const TMap<FString, FString>& Data)
{
	...
	std::string TypeSrc = (const char*)TCHAR_TO_ANSI(*EventType);
	std::string MessageSrc = (const char*)TCHAR_TO_ANSI(*EventMessage);

	FSpatialGDKSpanId SpanId = EventTracer->TraceEvent(TypeSrc.c_str(), MessageSrc.c_str(), ...
```

Reply:
Additionally I don't think we are getting much out of std::string? Why not use TCHAR_TO_ANSI directly rather than converting to std::string we don't seem to need for anything else?

Reply:
Also string literals have static storage duration, so no runtime cost.

Reply:
Ahh, you are referring to USpatialEventTracerUserInterface::TraceEvent and not SpatialEventTracer::TraceEvent. We could offer an overloaded version of USpatialEventTracerUserInterface::TraceEvent which does not take a string arg. We should also offer an overload that takes const char* for c++ usage. 

I believe we were worried about the lifetime of TCHAR_TO_ANSI. I believe we had issues with it in the past.

Reply:
As you long as you bind the outcome of TCHAR_TO_ANSI to a new object you are fine if the docs are to be believed:
`/**
 * NOTE: The objects these macros declare have very short lifetimes. They are
 * meant to be used as parameters to functions. You cannot assign a variable
 * to the contents of the converted string as the object will go out of
 * scope and the string released.
 *
 * NOTE: The parameter you pass in MUST be a proper string, as the parameter
 * is typecast to a pointer. If you pass in a char, not char* it will compile
 * and then crash at runtime.
 *
 * Usage:
 *
 *		SomeApi(TCHAR_TO_ANSI(SomeUnicodeString));
 *
 *		const char* SomePointer = TCHAR_TO_ANSI(SomeUnicodeString); <--- Bad!!!
 */
`

Reply:
LGTM! Will give it a quick test.